src/test/evo_islock_tests.cpp

```diff
@@ -0,0 +1,102 @@
+#include <boost/test/unit_test.hpp>
+#include <hash.h>
+#include <llmq/instantsend.h>
+#include <primitives/transaction.h>
+#include <streams.h>
+#include <string_view>
+#include <uint256.h>
+
+// For constructing dummy outpoints using uint256S.
+#include <util/strencodings.h>
+
+BOOST_AUTO_TEST_SUITE(evo_islock_tests)
+
+uint256 CalculateRequestId(const std::vector<COutPoint>& inputs)
+{
+    CHashWriter hw(SER_GETHASH, 0);
+    hw << std::string_view("islock");
+    hw << inputs;
+    return hw.GetHash();
+}
+
+BOOST_AUTO_TEST_CASE(getrequestid)
+{
+    // Create an empty InstantSendLock
+    llmq::CInstantSendLock islock;
+
+    // Compute expected hash for an empty inputs vector.
+    // Note: CInstantSendLock::GetRequestId() serializes the prefix "islock"
+    // followed by the 'inputs' vector.
+    {
+        const uint256 expected = CalculateRequestId(islock.inputs);
+
+        BOOST_CHECK(islock.GetRequestId() == expected);
+    }
+
+    // Now add two dummy inputs to the lock
+    islock.inputs.clear();
+    // Construct two dummy outpoints (using uint256S for a dummy hash)
+    COutPoint op1(uint256::ONE, 0);
+    COutPoint op2(uint256::TWO, 1);
+    islock.inputs.push_back(op1);
+    islock.inputs.push_back(op2);
+
+    const uint256 expected = CalculateRequestId(islock.inputs);
+
+    BOOST_CHECK(islock.GetRequestId() == expected);
+}
+
+BOOST_AUTO_TEST_CASE(deserialize_instantlock_from_realdata2)
+{
+    // Expected values from the provided getislocks output:
+    const std::string_view expectedTxidStr = "7b33968effa613e8ea9c1b5734c9bbbe467ff4650f8060caf8a5c213c6059d5b";
+    const std::string_view expectedCycleHashStr = "000000000000000bbd0b1bb95540351e7ee99c5b08efde076b3d712a57ea74d6";
+    const std::string_view expectedSignatureStr = "997d0b36738a9eef46ceeb4405998ff7235317708f277402799ffe05258015cae9b6bae"
+                                               "43683f992b2f50f70f8f0cb9c0f26af340b00903e93995c1345d1b2c5b697ebecdbe581"
+                                               "1dd112e11889101dcb4553b2bc206ab304026b96c07dec4f24";
```

Comment:
nit: clang-format complains, my bad 🙈 
```suggestion
    const std::string_view expectedSignatureStr =
        "997d0b36738a9eef46ceeb4405998ff7235317708f277402799ffe05258015cae9b6bae"
        "43683f992b2f50f70f8f0cb9c0f26af340b00903e93995c1345d1b2c5b697ebecdbe581"
        "1dd112e11889101dcb4553b2bc206ab304026b96c07dec4f
```